cmd/access/node_builder/unstaked_access_node_builder.go

```diff
@@ -20,109 +29,215 @@ func NewUnstakedAccessNodeBuilder(anb *FlowAccessNodeBuilder) *UnstakedAccessNod
+// These are the identities of the staked and unstaked ANs also acting as the DHT bootstrap server
+func (builder *FlowAccessNodeBuilder) deriveBootstrapPeerIdentities() {
+	// if bootstrap identities already provided (as part of alternate initialization as a library the skip reading command
+	// line params)
+	if builder.bootstrapIdentities != nil {
+		return
 	}
+	ids, err := BootstrapIdentities(builder.bootstrapNodeAddresses, builder.bootstrapNodePublicKeys)
+	builder.MustNot(err)
+	builder.bootstrapIdentities = ids
+}
 
-	if len(builder.bootstrapNodeAddresses) != len(builder.bootstrapNodePublicKeys) {
-		builder.Logger.Fatal().Msg("number of bootstrap node addresses and public keys should match")
+func (anb *UnstakedAccessNodeBuilder) validateParams() {
+	if anb.BaseConfig.BindAddr == cmd.NotSet || anb.BaseConfig.BindAddr == "" {
+		anb.Logger.Fatal().Msg("bind address not specified")
+	}
+	if anb.AccessNodeConfig.NetworkKey == nil {
+		anb.Logger.Fatal().Msg("networking key not provided")
+	}
+	if len(anb.bootstrapIdentities) > 0 {
+		return
+	}
+	if len(anb.bootstrapNodeAddresses) == 0 {
+		anb.Logger.Fatal().Msg("no bootstrap node address provided")
+	}
+	if len(anb.bootstrapNodeAddresses) != len(anb.bootstrapNodePublicKeys) {
+		anb.Logger.Fatal().Msg("number of bootstrap node addresses and public keys should match")
 	}
 }
 
-// deriveBootstrapPeerIdentities derives the Flow Identity of the bootstreap peers from the parameters.
-// These are the identity of the staked and unstaked AN also acting as the DHT bootstrap server
-func (builder *UnstakedAccessNodeBuilder) deriveBootstrapPeerIdentities() {
-	ids, err := BootstrapIdentities(builder.bootstrapNodeAddresses, builder.bootstrapNodePublicKeys)
+// initLibP2PFactory creates the LibP2P factory function for the given node ID and network key for the unstaked node.
+// The factory function is later passed into the initMiddleware function to eventually instantiate the p2p.LibP2PNode instance
+// The LibP2P host is created with the following options:
+// 		DHT as client and seeded with the given bootstrap peers
+// 		The specified bind address as the listen address
+// 		The passed in private key as the libp2p key
+//		No connection gater
+// 		No connection manager
+// 		Default libp2p pubsub options
+func (builder *UnstakedAccessNodeBuilder) initLibP2PFactory(ctx context.Context,
+	nodeID flow.Identifier,
+	networkKey crypto.PrivateKey) (p2p.LibP2PFactoryFunc, error) {
+
+	// the unstaked nodes act as the DHT clients
+	dhtOptions := []dht.Option{p2p.AsServer(false)}
+
+	// seed the DHT with the boostrap identities
+	bootstrapPeersOpt, err := p2p.WithBootstrapPeers(builder.bootstrapIdentities)
 	builder.MustNot(err)
-	builder.bootstrapIdentites = ids
+	dhtOptions = append(dhtOptions, bootstrapPeersOpt)
+
+	connManager := p2p.NewConnManager(builder.Logger, builder.Metrics.Network, p2p.TrackUnstakedConnections(builder.IdentityProvider))
```

Comment:
@vishalchangrani we may want to open a follow-up issue to do what you had in mind, i.e. making Stream establishment non-dependent on `Protect` calls for nodes that don't have connection gating